src/nft/components/sell/modal/ListingButton.tsx

```diff
@@ -0,0 +1,256 @@
+      : ''
+    return [disableListButton, warningMessage]
+  }, [
+    noMarketplacesSelected,
+    missingExpiration,
+    invalidExpiration,
+    overMaxExpiration,
+    listingsMissingPrice,
+    invalidPrices,
+    listingsBelowFloor,
+    listingsAboveSellOrderFloor,
+  ])
+
+  useEffect(() => {
+    setShowWarning(false)
+  }, [warningMessage])
+
+  const addWarningMessages = () => {
+    removeAllMarketplaceWarnings()
+    if (!missingExpiration && !noMarketplacesSelected) {
+      if (listingsMissingPrice.length > 0) {
+        for (const [asset, listing] of listingsMissingPrice) {
+          addMarketplaceWarning(asset, {
+            message: 'PLEASE SET A PRICE',
+            marketplace: listing.marketplace,
+          })
+        }
+      } else if (invalidPrices.length > 0) {
+        for (const [asset, listing] of invalidPrices) {
+          !listing.overrideFloorPrice &&
+            addMarketplaceWarning(asset, {
+              message: `INVALID PRICE`,
+              marketplace: listing.marketplace,
+            })
+        }
+      }
+    }
+    setShowWarning(true)
+  }
+
+  const warningWrappedClick = () => {
+    if ((!disableListButton && canContinue) || showWarningOverride) onClick()
+    else addWarningMessages()
+  }
+
+  return (
+    <Box style={{ position: 'relative' }} width="full">
+      {!showWarningOverride && showWarning && warningMessage.length > 0 && (
+        <Row
+          className={`${bodySmall} ${styles.warningTooltip}`}
+          transition="250"
+          onClick={() => setShowWarning(false)}
+          color="darkGray"
+          zIndex="3"
+          borderRadius="4"
+          backgroundColor="white"
+          height={!disableListButton ? '64' : '36'}
+          maxWidth="276"
+          position="absolute"
+          left="24"
+          bottom="52"
+          flexWrap={!disableListButton ? 'wrap' : 'nowrap'}
+          style={{ maxWidth: !disableListButton ? '225px' : '' }}
+          ref={warningRef}
+        >
+          <HazardIcon />
+          <Box marginLeft="4" marginRight="8">
+            {warningMessage}
+          </Box>
+          {!!disableListButton ? (
+            <Box paddingTop="6">
+              <XMarkIcon fill="darkGray" height="20" width="20" />
+            </Box>
+          ) : (
+            <Row
+              marginLeft="72"
+              cursor="pointer"
+              color="genieBlue"
+              onClick={() => {
+                setShowWarning(false)
+                setCanContinue(true)
+                onClick()
+              }}
+            >
+              Continue
+              <ArrowRightIcon height="20" width="20" />
+            </Row>
+          )}
+        </Row>
+      )}
+      <Box
+        as="button"
+        border="none"
+        backgroundColor="genieBlue"
+        cursor={
+          listingStatus === ListingStatus.APPROVED ||
```

Comment:
stuff like this might be more readable 

```[ListingStatus.APPROVED, ListingStatus.PENDING, ListingStatus.SIGNING].includes(listingStatus)```

It's used a few times throughout the pr

Reply:
I went with the rule of thumb of 3 or more, use an include array, otherwise stick with a basic comparison